pkg/controller/jobframework/tas_validation.go

```diff
@@ -25,9 +26,15 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	kueuebeta "sigs.k8s.io/kueue/apis/kueue/v1beta1"
 )
 
-func ValidateTASPodSetRequest(replicaPath *field.Path, replicaMetadata *metav1.ObjectMeta) field.ErrorList {
+func ValidateTASPodSetRequestForJob(job GenericJob, replicaMetaPath *field.Path, replicaMetadata *metav1.ObjectMeta) field.ErrorList {
+	podSets, _ := job.PodSets()
```

Comment:
Make sure this is called only when TopologyAwareScheduling feature gate is enabled

Reply:
Done, now the validations for TAS run only when TAS is enabled